src/libAtomVM/mailbox.c

```diff
@@ -40,22 +38,50 @@ void mailbox_init(Mailbox *mbx)
     mbx->receive_pointer_prev = NULL;
 }
 
-// Generic destroy function name refers to signal message to make sure ports do
-// not call it. They should call mailbox_remove instead.
-void mailbox_destroy_signal_message(MailboxMessage *m)
+// Convert a mailbox message (struct Message or struct TermSignal) to a heap
+// fragment (HeapFragment) so it can be owned by the recipient.
+// We assert this layout mapping is correct.
+_Static_assert(offsetof(struct Message, base) + offsetof(struct MailboxMessage, next) == offsetof(HeapFragment, next) ? 1 : 0, "Message.base.next doesn't match HeapFragment.next");
```

Comment:
Looks fine, my only suggestion here is trying to use `clang-format --lines=41:69` with ColumnWidth set to 100. It might look better.

Reply:
ColumnLimit is currently set to 0 and 100 doesn't look very good IMHO.
I did split all lines manually and `clang-format` (with `ColumnLimit` as 0) seems happy.